authors/apps/comments/views.py

```diff
@@ -0,0 +1,161 @@
+        """
+        This function updates a given comment
+        for an article with given id and slag
+        """
+        comment = get_object_or_404(Comment, pk=pk)
+        article = get_object_or_404(Article, pk=comment.article.pk)
+        user = Profile.objects.get(user=request.user)
+        data = request.data.get("comment")
+
+        if comment.commented_by != user:
+            return Response({
+                "message": "You can only edit a comment you created"
+            }, status=status.HTTP_403_FORBIDDEN)
+        serializer = self.serializer_class(
+            comment,
+            data,
+            partial=True, context={"article": article})
+        serializer.is_valid(raise_exception=True)
+        if comment.body == data["body"]:
+            return Response({
+                "message": "Please ensure the comment body has changed"
+            }, status=status.HTTP_400_BAD_REQUEST)
+        serializer.save()
+        return Response({
+            "comment":serializer.data,
+            "message": "comment successfully updated"
+        }, status=status.HTTP_200_OK)
+
+class CommentLikeView(generics.GenericAPIView):
+    serializer_class = CommentLikeSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (CommentJSONRenderer, )
+
+    def put(self, *args, **kwargs):
+        """
+        function for liking a comment on a given article
+        """
+        comment = get_object_or_404(Comment, pk=kwargs.get("pk"))
+        user = get_object_or_404(Profile, user=self.request.user)
+        try:
+            CommentLike.objects.get(liked_by=user)
+        except CommentLike.DoesNotExist:
+            serializer = self.serializer_class(data={})
+            serializer.is_valid(raise_exception=True)
+            serializer.save(liked_by=user,
+                            comment=comment, like_status=True)
+            return Response({
+                "message": "comment liked successfully"
+            }, status=status.HTTP_200_OK)
+
+        return Response({
+            "message": "you already liked this comment"
+        }, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, *args, **kwargs):
+        """
+        Function to retrieve all likes on a comment
+        """
+        comment = get_object_or_404(Comment, pk=kwargs.get("pk"))
+        likes = CommentLike.objects.filter(
+            like_status=True).filter(comment=comment)
+        serializer = self.serializer_class(likes, many=True)
+        return Response({
+            "likes": serializer.data,
+            "likesCount": likes.count()
+        }, status=status.HTTP_200_OK)
+
+    def delete(self, *args, **kwargs):
+        """
+        Function to remove a like from a comment you
+        had already liked
+        """
+        try:
+            user = get_object_or_404(
+                Profile, user=self.request.user)
+            comment = get_object_or_404(Comment, pk=kwargs.get("pk"))
```

Comment:
Thanks for the good work @wasibani-roy, however, I feel like this line is not needed as you don't seem to use it. please clarify if you use it somewhere in the implementation. If you don't use it, we can remove it to improve the performance of the database.

Reply:
This try is there to handle situations where the user may try to unlike a comment they had not originally like you see such if not handled would cause the application to crash. That's why we have the try-catch block there